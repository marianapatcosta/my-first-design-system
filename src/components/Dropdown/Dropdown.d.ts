import React, { ChangeEvent, CSSProperties } from 'react';
export interface DropdownProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * Dropdown's label
     */
    label?: string;
    /**
     * Dropdown's name
     */
    name?: string;
    /**
     * Dropdown's displaying text when no option is selected
     */
    placeholder?: string;
    /**
     * Is Dropdown disabled?
     */
    disabled?: boolean;
    /**
     * the option that is selected
     */
    selectedOption?: any;
    /**
     * name of the field to be displayed in the list of dropdown's options, when options prop is an array of objects
     */
    optionKey?: string;
    /**
     * options to display in the dropdown content
     */
    options: Array<any>;
    /**
     * Change handler
     */
    onChooseOption: (event?: ChangeEvent<HTMLSelectElement>) => void;
}
export declare const Dropdown: React.FC<DropdownProps>;
