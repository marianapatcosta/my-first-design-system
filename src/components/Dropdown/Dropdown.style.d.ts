export declare const StyledDropdownContainer: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const StyledDropdown: import("styled-components").StyledComponent<"div", any, {
    isExpanded: boolean;
    disabled?: boolean | undefined;
}, never>;
export declare const StyledDropdownLabel: import("styled-components").StyledComponent<"label", any, {
    disabled?: boolean | undefined;
}, never>;
export declare const StyledDropdownHeader: import("styled-components").StyledComponent<"select", any, {}, never>;
export declare const StyledDropdownOption: import("styled-components").StyledComponent<"option", any, {}, never>;
