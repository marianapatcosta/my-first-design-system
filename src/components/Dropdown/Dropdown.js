var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState, } from 'react';
import { Chevron } from '../../assets/icons';
import { KEYBOARD_KEY } from '../../constants';
import { StyledDropdown, StyledDropdownContainer, StyledDropdownLabel, StyledDropdownHeader, StyledDropdownOption, } from './Dropdown.style';
export var Dropdown = function (_a) {
    var className = _a.className, name = _a.name, placeholder = _a.placeholder, options = _a.options, selectedOption = _a.selectedOption, label = _a.label, disabled = _a.disabled, optionKey = _a.optionKey, onChooseOption = _a.onChooseOption;
    var _b = useState(false), isExpanded = _b[0], setIsExpanded = _b[1];
    var handleOptionChange = function (event) {
        event.stopPropagation();
        onChooseOption(event);
    };
    var handleKeyEvent = function (event) {
        event.stopPropagation();
        // keyUp
        if (event.key === KEYBOARD_KEY.ESCAPE_KEY) {
            return setIsExpanded(false);
        }
        // keyDown
        if (event.key === KEYBOARD_KEY.ENTER_KEY) {
            return setIsExpanded(function (prevIsExpanded) { return !prevIsExpanded; });
        }
        // keyUp
        if (event.key === KEYBOARD_KEY.SPACE_KEY) {
            return setIsExpanded(true);
        }
    };
    return (_jsxs(StyledDropdownContainer, __assign({ className: className }, { children: [!!label && (_jsx(StyledDropdownLabel, __assign({ htmlFor: name || label, disabled: disabled }, { children: label }), void 0)),
            _jsxs(StyledDropdown, __assign({ isExpanded: isExpanded, disabled: disabled }, { children: [_jsx("img", { src: Chevron, alt: 'chevron-icon' }, void 0),
                    _jsxs(StyledDropdownHeader, __assign({ "aria-label": "Select " + (name || label), name: name || label, disabled: disabled, value: selectedOption, onChange: handleOptionChange, onClick: function () { return setIsExpanded(function (prevIsExpanded) { return !prevIsExpanded; }); }, onBlur: function () { return setIsExpanded(false); }, onKeyUp: handleKeyEvent, onKeyDown: handleKeyEvent }, { children: [!!placeholder && (_jsx(StyledDropdownOption, __assign({ value: '', hidden: true }, { children: placeholder }), void 0)),
                            options.map(function (option, index) {
                                var optionValue = optionKey ? option[optionKey] : option;
                                return (_jsx(StyledDropdownOption, __assign({ "data-testid": 'select-option', value: optionValue }, { children: optionValue }), "dropdown-option-" + (index + Math.random())));
                            })] }), void 0)] }), void 0)] }), void 0));
};
Dropdown.defaultProps = {
    id: '',
    className: '',
    style: {},
    name: '',
    label: '',
    placeholder: '',
    disabled: false,
    options: [],
    selectedOption: null,
    optionKey: '',
    onChooseOption: function () { return null; },
};
