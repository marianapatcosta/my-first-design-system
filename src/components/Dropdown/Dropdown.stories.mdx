import { Meta } from '@storybook/addon-docs/blocks';
import { Dropdown } from './Dropdown';

<Meta title="Design System|Dropdown" component={Dropdown} />

# Dropdown
Simple Dropdown component based on HTML select element

## Usage
Use dropdown when there are more than two options available and the user should pick one of them.
The options can be primitives or objects. optionKey prop indicates which field should be displayed
in dropdown options and selected option when options are of type Object

- Default Dropdown:
  <Story name='Default'>
    <Dropdown 
      placeholder='select an option' 
      options={['option 1', 'option 2', 'option 3']}
      selectedOption={null}
      />
  </Story>

- Dropdown With label:
  <Story name='WithLabel'>
     <Dropdown 
      placeholder='select an option' 
      options={['option 1', 'option 2', 'option 3']}
      selectedOption={null}
      label='Example'
      />
  </Story>

## Other important notes
not applicable.
