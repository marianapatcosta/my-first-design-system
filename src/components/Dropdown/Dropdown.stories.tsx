import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Dropdown, DropdownProps } from './Dropdown';

export default {
  title: 'Dropdown',
  component: Dropdown,
  args: {
    options: ['option 1', 'option 2', 'option 3'],
    selectedOption: '',
    placeholder: 'select an option',
  },
} as Meta;

const Template: Story<DropdownProps> = (args) => <Dropdown {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const WithLabel = Template.bind({});
WithLabel.args = {label: 'Example'};
