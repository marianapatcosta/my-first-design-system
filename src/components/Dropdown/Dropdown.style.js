var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles, disabledStyles, colors } from '../../styles';
export var StyledDropdownContainer = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n  display: flex;\n  flex-direction: column;\n  width: 100%;\n  max-width: 10rem;\n  position: relative;\n"], ["\n  ", "\n  display: flex;\n  flex-direction: column;\n  width: 100%;\n  max-width: 10rem;\n  position: relative;\n"])), baseStyles);
export var StyledDropdown = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  display: flex;\n  width: 100%;\n  position: relative;\n\n  img {\n    width: 0.8rem;\n    position: absolute;\n    top: 50%;\n    bottom: 50%;\n    right: 0.6rem;\n    margin: auto 0;\n    z-index: 1;\n    ", ";\n    -webkit-transition: transform 0.5s linear;\n    transition: transform 0.5s linear;\n\n    ", "\n\n    ", "\n  }\n  \n  /* :after {\n    position: absolute;\n    top: 50%;\n    bottom: 50%;\n    right: 0.8rem;\n    margin: auto 0;\n    content: '';\n    pointer-events: none;\n    width: 0.4em;\n    height: 0.4em;\n    border-right: 0.125rem solid ", ";\n    border-bottom: 0.125rem solid ", ";\n    transform: rotate(45deg);\n    -webkit-transition: transform 0.5s linear;\n    transition: transform 0.5s linear;\n\n    ", "\n  }*/\n"], ["\n  display: flex;\n  width: 100%;\n  position: relative;\n\n  img {\n    width: 0.8rem;\n    position: absolute;\n    top: 50%;\n    bottom: 50%;\n    right: 0.6rem;\n    margin: auto 0;\n    z-index: 1;\n    ", ";\n    -webkit-transition: transform 0.5s linear;\n    transition: transform 0.5s linear;\n\n    ", "\n\n    ",
    "\n  }\n  \n  /* :after {\n    position: absolute;\n    top: 50%;\n    bottom: 50%;\n    right: 0.8rem;\n    margin: auto 0;\n    content: '';\n    pointer-events: none;\n    width: 0.4em;\n    height: 0.4em;\n    border-right: 0.125rem solid ", ";\n    border-bottom: 0.125rem solid ", ";\n    transform: rotate(45deg);\n    -webkit-transition: transform 0.5s linear;\n    transition: transform 0.5s linear;\n\n    ", "\n  }*/\n"])), colors.icon, function (_a) {
    var isExpanded = _a.isExpanded;
    return isExpanded && "transform: rotate(180deg);";
}, function (_a) {
    var disabled = _a.disabled;
    return disabled &&
        " border-bottom-color: " + (disabled && colors.disabled) + ";\n      border-right-color: " + (disabled && colors.disabled) + ";\n      ";
}, colors.highlight, colors.highlight, function (_a) {
    var isExpanded = _a.isExpanded;
    return isExpanded && "transform: rotate(225deg);";
});
export var StyledDropdownLabel = styled.label(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  color: ", ";\n  margin: 0 0.8rem 0.3rem 0.3rem;\n  font-size: 95%;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n\n  ", "\n"], ["\n  color: ", ";\n  margin: 0 0.8rem 0.3rem 0.3rem;\n  font-size: 95%;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n\n  ", "\n"])), colors.highlight, function (_a) {
    var disabled = _a.disabled;
    return disabled && "" + disabledStyles;
});
export var StyledDropdownHeader = styled.select(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  // reset of styles, including removing the default dropdown arrow\n  appearance: none;\n  border: none;\n  margin: 0;\n  width: 100%;\n  font-family: inherit;\n  line-height: inherit;\n  ", "\n  background-color: ", ";\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  padding: 0.3rem 2rem 0.3rem 0.8rem;\n  font-size: 95%;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n  position: relative;\n  cursor: pointer;\n \n\n  ::-ms-expand {\n    display: none;\n  }\n\n  ", "\n"], ["\n  // reset of styles, including removing the default dropdown arrow\n  appearance: none;\n  border: none;\n  margin: 0;\n  width: 100%;\n  font-family: inherit;\n  line-height: inherit;\n  ", "\n  background-color: ", ";\n  box-shadow: 0 0.063rem 0.25rem ", ";\n  padding: 0.3rem 2rem 0.3rem 0.8rem;\n  font-size: 95%;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n  position: relative;\n  cursor: pointer;\n \n\n  ::-ms-expand {\n    display: none;\n  }\n\n  ", "\n"])), baseStyles, colors.primary, colors.shadow, function (_a) {
    var disabled = _a.disabled;
    return disabled && "" + disabledStyles;
});
export var StyledDropdownOption = styled.option(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  padding: 0.625rem 0.3rem;\n  line-height: 4rem;\n  box-sizing: border-box;\n  cursor: pointer;\n\n  :hover {\n    background-color: ", " !important;\n  }\n\n  :checked {\n    font-weight: bold;\n  }\n"], ["\n  padding: 0.625rem 0.3rem;\n  line-height: 4rem;\n  box-sizing: border-box;\n  cursor: pointer;\n\n  :hover {\n    background-color: ", " !important;\n  }\n\n  :checked {\n    font-weight: bold;\n  }\n"])), colors.secondary);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
