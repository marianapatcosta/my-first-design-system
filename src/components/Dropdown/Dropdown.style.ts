import styled from 'styled-components';
import { baseStyles, disabledStyles, colors } from '../../styles';

export const StyledDropdownContainer = styled.div`
  ${baseStyles}
  display: flex;
  flex-direction: column;
  width: 100%;
  max-width: 10rem;
  position: relative;
`;

export const StyledDropdown = styled.div<{
  isExpanded: boolean;
  disabled?: boolean;
}>`
  display: flex;
  width: 100%;
  position: relative;

  img {
    width: 0.8rem;
    position: absolute;
    top: 50%;
    bottom: 50%;
    right: 0.6rem;
    margin: auto 0;
    z-index: 1;
    filter: ${colors.icon};
    -webkit-transition: transform 0.5s linear;
    transition: transform 0.5s linear;

    ${({ isExpanded }) => isExpanded && `transform: rotate(180deg);`}

    ${({ disabled }) =>
      disabled &&
      ` border-bottom-color: ${disabled && colors.disabled};
      border-right-color: ${disabled && colors.disabled};
      `}
  }

  /* :after {
    position: absolute;
    top: 50%;
    bottom: 50%;
    right: 0.8rem;
    margin: auto 0;
    content: '';
    pointer-events: none;
    width: 0.4em;
    height: 0.4em;
    border-right: 0.125rem solid ${colors.highlight};
    border-bottom: 0.125rem solid ${colors.highlight};
    transform: rotate(45deg);
    -webkit-transition: transform 0.5s linear;
    transition: transform 0.5s linear;

    ${({ isExpanded }) => isExpanded && `transform: rotate(225deg);`}
  }*/
`;

export const StyledDropdownLabel = styled.label<{ disabled?: boolean }>`
  color: ${colors.highlight};
  margin: 0 0.8rem 0.3rem 0.3rem;
  font-size: 95%;
  white-space: nowrap;
  text-overflow: ellipsis;

  ${({ disabled }) => disabled && `${disabledStyles}`}
`;

export const StyledDropdownHeader = styled.select`
  // reset of styles, including removing the default dropdown arrow
  appearance: none;
  border: none;
  margin: 0;
  width: 100%;
  font-family: inherit;
  line-height: inherit;
  ${baseStyles}
  background-color: ${colors.primary};
  box-shadow: 0 0.063rem 0.25rem ${colors.shadow};
  padding: 0.3rem 2rem 0.3rem 0.8rem;
  font-size: 95%;
  white-space: nowrap;
  text-overflow: ellipsis;
  position: relative;
  cursor: pointer;

  ::-ms-expand {
    display: none;
  }

  ${({ disabled }) => disabled && `${disabledStyles}`}
`;

export const StyledDropdownOption = styled.option`
  padding: 0.625rem 0.3rem;
  line-height: 4rem;
  box-sizing: border-box;
  cursor: pointer;

  :hover {
    background-color: ${colors.secondary} !important;
  }

  :checked {
    font-weight: bold;
  }
`;
