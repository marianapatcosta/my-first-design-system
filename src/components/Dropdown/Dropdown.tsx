import React, { ChangeEvent, KeyboardEvent, useState } from 'react';
import { KEYBOARD_KEYS } from '../../constants';
import {
  StyledDropdown,
  StyledDropdownContainer,
  StyledDropdownLabel,
  StyledDropdownHeader,
  StyledDropdownOption,
} from './Dropdown.style';

export interface DropdownProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * Dropdown's label
   */
  label?: string;
  /**
   * Dropdown's name
   */
  name?: string;
  /**
   * Dropdown's displaying text when no option is selected
   */
  placeholder?: string;
  /**
   * Is Dropdown disabled?
   */
  disabled?: boolean;
  /**
   * the option that is selected
   */
  selectedOption: any;
  /**
   * name of the field to be displayed in the list of dropdown's options, when options prop is an array of objects
   */
  optionKey?: string;
  /**
   * options to display in the dropdown content
   */
  options: Array<any>;
  /**
   * Change handler
   */
  onChooseOption: (event?: ChangeEvent<HTMLSelectElement>) => void;
}

export const Dropdown: React.FC<DropdownProps> = ({
  className,
  name,
  placeholder,
  options,
  selectedOption,
  label,
  disabled,
  optionKey,
  onChooseOption,
}) => {
  const [isExpanded, setIsExpanded] = useState(false);
  const handleOptionChange = (event: ChangeEvent<HTMLSelectElement>) => {
    event.stopPropagation();
    onChooseOption(event);
  };

  const handleKeyEvent = (event: KeyboardEvent<HTMLSelectElement>) => {
    event.stopPropagation();
    if (event.key === KEYBOARD_KEYS.ESCAPE_KEY) {
      return setIsExpanded(false);
    }
    if (event.key === KEYBOARD_KEYS.ENTER_KEY) {
      return setIsExpanded((prevIsExpanded) => !prevIsExpanded);
    }
    if (event.key === KEYBOARD_KEYS.SPACE_KEY) {
      return setIsExpanded(true);
    }
  };

  return (
    <StyledDropdownContainer className={className}>
      {!!label && (
        <StyledDropdownLabel htmlFor={name || label} disabled={disabled}>
          {label}
        </StyledDropdownLabel>
      )}
      <StyledDropdown isExpanded={isExpanded} disabled={disabled}>
        <StyledDropdownHeader
          aria-label={`Select ${name || label}`}
          name={name || label}
          disabled={disabled}
          value={selectedOption}
          onChange={handleOptionChange}
          onClick={() => setIsExpanded((prevIsExpanded) => !prevIsExpanded)}
          onBlur={() => setIsExpanded(false)}
          onKeyUp={handleKeyEvent}
          onKeyDown={handleKeyEvent}
        >
          {!!placeholder && (
            <StyledDropdownOption value='' hidden>
              {placeholder}
            </StyledDropdownOption>
          )}
          {options.map((option: any, index: number) => {
            const optionLabel = optionKey ? option[optionKey] : option;
            return (
              <StyledDropdownOption
                key={`dropdown-option-${index + Math.random()}`}
                value={option}
              >
                {optionLabel}
              </StyledDropdownOption>
            );
          })}
        </StyledDropdownHeader>
      </StyledDropdown>
    </StyledDropdownContainer>
  );
};

Dropdown.defaultProps = {
  className: '',
  id: '',
  name: '',
  label: '',
  placeholder: '',
  disabled: false,
  options: [],
  selectedOption: null,
  optionKey: '',
  onChooseOption: () => null,
};
