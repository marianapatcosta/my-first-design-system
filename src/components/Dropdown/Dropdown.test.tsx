import React from 'react';
import { render, getByText, fireEvent } from '@testing-library/react';
import { Dropdown } from './Dropdown';

describe('Dropdown', () => {
  const defaultProps = {
    options: ['option 1', 'option 2', 'option 3'],
    selectedOption: '',
    placeholder: 'select an option',
    onChooseOption: () => null,
  };
  test('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Dropdown {...defaultProps} />);

    expect(firstChild).toMatchSnapshot();
  });

  test('should display label if label prop is passed', () => {
    const { container } = render(
      <Dropdown {...defaultProps} label='Dropdown' />
    );
    getByText(container, 'Dropdown');
  });

  test('should display the field of `optionKey` as option element label, if objects are passed in `options array', () => {
    const { getByText } = render(
      <Dropdown
        {...defaultProps}
        label='Dropdown'
        options={[
          { label: 'option 1' },
          { label: 'option 2' },
          { label: 'option 3' },
        ]}
        optionKey='label'
      />
    );
    getByText('option 1');
  });

  test('should return null on choose option if onChooseOption prop is not passed', () => {
    const result =
      Dropdown.defaultProps?.onChooseOption &&
      Dropdown.defaultProps.onChooseOption();
    expect(result).toBeNull();
  });

  test('should handle change events and simulate selection', async () => {
    const onChooseOptionMock = jest.fn();
    const { getByLabelText } = render(
      <Dropdown {...defaultProps} onChooseOption={onChooseOptionMock} />
    );

    const select = getByLabelText(/select/i) as HTMLSelectElement;
    fireEvent.change(select);
    expect(onChooseOptionMock).toBeCalled();

    /*  const select = getByLabelText(/select/i) as HTMLSelectElement;
    userEvent.selectOptions(select, ['option 2']);
    let options = (getAllByTestId(
      'select-option'
    ) as unknown) as HTMLOptionsCollection;
    expect(options[0].selected).toBe(false);
    expect(options[1].selected).toBe(true);
    expect(options[2].selected).toBeFalsy(); */
  });

  test('should set isExpanded = true onclick', () => {
    const { getByLabelText, getByAltText } = render(
      <Dropdown {...defaultProps} />
    );
    const select = getByLabelText(/select/i) as HTMLSelectElement;
    const chevron = getByAltText('chevron-icon');
    fireEvent.click(select);
    const style = window.getComputedStyle(chevron);
    expect(style.transform).toBe('rotate(180deg)');
  });

  test('should handle blur events and sets isExpanded = false', () => {
    const { getByLabelText, getByAltText } = render(
      <Dropdown {...defaultProps} />
    );
    const select = getByLabelText(/select/i) as HTMLSelectElement;
    const chevron = getByAltText('chevron-icon');
    fireEvent.change(select);
    fireEvent.blur(select);
    const style = window.getComputedStyle(chevron);
    expect(style.transform).toBe('');
  });

  describe('Keyboard events handling', () => {
    let select: HTMLSelectElement;
    let chevron: HTMLImageElement;
    beforeEach(() => {
      const { getByLabelText, getByAltText } = render(
        <Dropdown {...defaultProps} />
      );
      select = getByLabelText(/select/i) as HTMLSelectElement;
      chevron = getByAltText('chevron-icon') as HTMLImageElement;
    });
    test('should set isExpanded = true on `enter`', () => {
      fireEvent.keyDown(select, { key: 'Enter' });
      const style = window.getComputedStyle(chevron);
      expect(style.transform).toBe('rotate(180deg)');
    });

    test('should set isExpanded = true on `space`', () => {
      fireEvent.keyUp(select, { key: ' ' });
      const style = window.getComputedStyle(chevron);
      expect(style.transform).toBe('rotate(180deg)');
    });

    test('should set isExpanded = true on `escape`', () => {
      fireEvent.keyUp(select, { key: 'Escape' });
      const style = window.getComputedStyle(chevron);
      expect(style.transform).toBe('');
    });
  });

  describe('Disabled Dropdown', () => {
    test('should be disabled', () => {
      const { container } = render(<Dropdown {...defaultProps} disabled />);

      const dropdown = container.querySelector('select');
      expect(dropdown).toHaveAttribute('disabled');
    });

    test('should not handle change events', () => {
      const onChooseOptionMock = jest.fn();
      const {
        container: { firstChild },
      } = render(
        <Dropdown
          {...defaultProps}
          disabled={true}
          onChooseOption={onChooseOptionMock}
        />
      );

      firstChild && fireEvent.click(firstChild);
      expect(onChooseOptionMock).not.toBeCalled();
    });
  });
});
