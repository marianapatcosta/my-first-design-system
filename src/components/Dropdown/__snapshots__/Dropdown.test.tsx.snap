// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ gmhHJm"
>
  <div
    class="sc-gsTCUz hLxDNj"
  >
    <select
      aria-label="Select "
      class="sc-hKgILt hxRLa-D"
      name=""
    >
      <option
        class="sc-eCssSg kgbcsp"
        hidden=""
        value=""
      >
        select an option
      </option>
      <option
        class="sc-eCssSg kgbcsp"
        value="option 1"
      >
        option 1
      </option>
      <option
        class="sc-eCssSg kgbcsp"
        value="option 2"
      >
        option 2
      </option>
      <option
        class="sc-eCssSg kgbcsp"
        value="option 3"
      >
        option 3
      </option>
    </select>
  </div>
</div>
`;
